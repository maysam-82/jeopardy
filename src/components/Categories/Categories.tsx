import React from 'react';
import { ListGroup } from 'react-bootstrap';
import { ICategory } from '../../types/category';

interface ICategories {
	categories: ICategory[];
	categoryClick: (id: number) => void;
}

const Categories: React.SFC<ICategories> = ({ categories, categoryClick }) => {
	return (
		<ListGroup>
			{categories.map(({ id, title }) => {
				return (
					<ListGroup.Item
						key={id}
						action
						onClick={() => categoryClick(id)}
						variant="secondary"
					>
						{title}
					</ListGroup.Item>
				);
			})}
		</ListGroup>
	);
};

export default Categories;
