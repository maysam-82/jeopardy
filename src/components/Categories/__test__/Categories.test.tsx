import React from 'react';
import { shallow } from 'enzyme';
import Categories, { ICategoriesProps } from '../Categories';
import { testCategories } from '../../../fixtures/testData';

const testCategoryProps: ICategoriesProps = {
	categories: testCategories,
	categoryClick: jest.fn(),
};

describe('Categories component', () => {
	const component = shallow(<Categories {...testCategoryProps} />);
	it('should render Categories', () => {
		expect(component).toMatchSnapshot();
	});
	it('should render `ListGroupItem` component', () => {
		expect(component.find('ListGroupItem').length).toEqual(3);
	});
	describe('When user clicks on Category', () => {
		beforeEach(() => {
			component.find('ListGroupItem').at(1).simulate('click');
		});
		it('should call categoryClick', () => {
			expect(testCategoryProps.categoryClick).toHaveBeenCalledTimes(1);
		});
	});
});
