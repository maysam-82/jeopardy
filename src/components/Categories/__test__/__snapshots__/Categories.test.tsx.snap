// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component should render Categories 1`] = `
<ListGroup
  horizontal={null}
  variant={null}
>
  <ListGroupItem
    action={true}
    active={false}
    disabled={false}
    key="11531"
    onClick={[Function]}
    variant="secondary"
  >
    mixed bag
  </ListGroupItem>
  <ListGroupItem
    action={true}
    active={false}
    disabled={false}
    key="11532"
    onClick={[Function]}
    variant="secondary"
  >
    let's "ch"at
  </ListGroupItem>
  <ListGroupItem
    action={true}
    active={false}
    disabled={false}
    key="5412"
    onClick={[Function]}
    variant="secondary"
  >
    prehistoric times
  </ListGroupItem>
</ListGroup>
`;
